src/app/loadout-builder/LoadoutBuilder.tsx

```diff
@@ -128,278 +243,205 @@ function mapStateToProps() {
-      new Set(statKeys.filter((statType) => !statFilters[statType].ignored))
+function LoadoutBuilder({
+  account,
+  storesLoaded,
+  stores,
+  isPhonePortrait,
+  items,
+  defs,
+  searchConfig,
+  filters,
+  location,
+}: Props) {
+  // Memoizing to ensure these functions are only created once
+  const [filterItemsMemoized, filterSetsMemoized, processMemoized, getEnabledStats] = useMemo(
+    () => [
+      memoizeOne(filterItems),
+      memoizeOne(filterGeneratedSets),
+      memoizeOne(process),
+      memoizeOne(
+        (statFilters: Readonly<{ [statType in StatTypes]: MinMaxIgnored }>) =>
+          new Set(statKeys.filter((statType) => !statFilters[statType].ignored))
+      ),
+    ],
+    []
   );
 
-  constructor(props: Props) {
-    super(props);
-
-    let lockedMap: LockedMap = {};
-    const { stores, location } = this.props;
-    if (stores.length && location.state?.loadout) {
-      for (const loadoutItem of location.state.loadout.items) {
-        if (loadoutItem.equipped) {
-          const item = getItemAcrossStores(stores, loadoutItem);
-          if (item && isLoadoutBuilderItem(item)) {
-            lockedMap = {
-              ...lockedMap,
-              [item.bucket.hash]: addLockedItem(
-                { type: 'item', item, bucket: item.bucket },
-                lockedMap[item.bucket.hash]
-              ),
-            };
-          }
-        }
-      }
-    }
-
-    this.state = {
+  const [
+    {
       lockedMap,
-      statFilters: {
-        Mobility: { min: 0, max: 10, ignored: false },
-        Resilience: { min: 0, max: 10, ignored: false },
-        Recovery: { min: 0, max: 10, ignored: false },
-        Discipline: { min: 0, max: 10, ignored: false },
-        Intellect: { min: 0, max: 10, ignored: false },
-        Strength: { min: 0, max: 10, ignored: false },
-      },
-      lockedSeasonalMods: [],
-      lockedArmor2Mods: {
-        [ModPickerCategories.general]: [],
-        [ModPickerCategories.helmet]: [],
-        [ModPickerCategories.gauntlets]: [],
-        [ModPickerCategories.chest]: [],
-        [ModPickerCategories.leg]: [],
-        [ModPickerCategories.classitem]: [],
-        [ModPickerCategories.seasonal]: [],
-      },
-      minimumPower: 750,
-      query: '',
-      statOrder: statKeys,
-      selectedStoreId: getCurrentStore(props.stores)?.id,
-      assumeMasterwork: false,
-    };
-  }
-
-  componentDidMount() {
-    this.subscriptions.add(
-      D2StoresService.getStoresStream(this.props.account).subscribe((stores) => {
+      lockedSeasonalMods,
+      lockedArmor2Mods,
+      selectedStoreId,
+      statFilters,
+      minimumPower,
+      query,
+      statOrder,
+      assumeMasterwork,
+    },
+    stateDispatch,
+  ] = useReducer(stateReducer, { stores, location }, init);
+
+  useEffect(() => {
+    const subscriptions = new Subscriptions();
+    subscriptions.add(
+      D2StoresService.getStoresStream(account).subscribe((stores) => {
```

Comment:
We have a `useSubscription` hook you can use for this.

Reply:
Will do